fe/fe-core/src/main/java/com/starrocks/scheduler/MvTaskRunProcessor.java

```diff
@@ -356,47 +404,174 @@ private void refreshMv(TaskRunContext context, MaterializedView materializedView
-        insertStmt.setOrigStmt(new OriginStatement(insertSql, 0));
         StmtExecutor executor = new StmtExecutor(ctx, insertStmt);
         ctx.setExecutor(executor);
         ctx.setThreadLocalInfo();
+        ctx.setStmtId(new AtomicInteger().incrementAndGet());
+        ctx.setExecutionId(UUIDUtil.toTUniqueId(ctx.getQueryId()));
         try {
-            executor.execute();
+            Map<Long, Map<String, MaterializedView.BasePartitionInfo>> selectedBasePartitionInfos = Maps.newHashMap();
+            database.readLock();
+            ExecPlan execPlan;
+            try {
+                execPlan = new StatementPlanner().plan(insertStmt, ctx);
+                List<ScanNode> scanNodes = execPlan.getScanNodes();
+                for (ScanNode scanNode : scanNodes) {
+                    if (scanNode instanceof OlapScanNode) {
+                        OlapScanNode olapScanNode = (OlapScanNode) scanNode;
+                        Map<String, MaterializedView.BasePartitionInfo> selectedPartitionIdVersions =
+                                getSelectedPartitionInfos(olapScanNode);
+                        OlapTable olapTable = olapScanNode.getOlapTable();
+                        selectedBasePartitionInfos.put(olapTable.getId(), selectedPartitionIdVersions);
+                    }
+                }
+                // check partition changed
+                checkPartitionChange(selectedBasePartitionInfos, plannedOlapTablePartitionInfos);
+            } finally {
+                database.readUnlock();
+            }
+            executor.handleDMLStmt(execPlan, insertStmt);
+            // use selected base partition info to update mv version map & partition name map
+            for (Map.Entry<Long, Map<String, MaterializedView.BasePartitionInfo>> longMapEntry :
+                    selectedBasePartitionInfos.entrySet()) {
+                long baseTableId = longMapEntry.getKey();
+                Map<String, MaterializedView.BasePartitionInfo> baseTableIdAndVersion = longMapEntry.getValue();
+                for (Map.Entry<String, MaterializedView.BasePartitionInfo> basicTablePartitionIdVersions :
+                        baseTableIdAndVersion.entrySet()) {
+                    String baseTablePartitionName = basicTablePartitionIdVersions.getKey();
+                    MaterializedView.BasePartitionInfo baseTablePartitionVersion =
+                            basicTablePartitionIdVersions.getValue();
+                    materializedView.updateBasePartition(baseTableId, baseTablePartitionName,
+                            baseTablePartitionVersion);
+                }
+            }
+            materializedView.getRefreshScheme().setLastRefreshTime(Utils.getLongFromDateTime(LocalDateTime.now()));
         } catch (Exception e) {
-            throw new SemanticException("Refresh materialized view failed:" + insertSql, e);
+            throw new SemanticException("Refresh materialized view failed: " + e.getMessage(), e);
```

Comment:
e contains the error message if you log the exception, why add e.getMessage to the message?